src/utils/mount.js

```diff
@@ -0,0 +1,99 @@
+    let existingFeathersConfig = require(feathersConfigPath);
+
+    // add new service to root config for bootstrapping
+    let serviceConfigPath = path.resolve(options.path, options.name, options.name + '.json');
+    let relativeServiceConfigPath = path.relative(feathersConfigDirname, serviceConfigPath);
+    let feathersConfigChanges = { use: { ['/' + options.name]: { require: './' + relativeServiceConfigPath } } };
+    let newFeathersConfig = merge(existingFeathersConfig, feathersConfigChanges);
+
+    debug(`Attempting to mount ${options.name} service to feathers at ${feathersConfigPath}`);
+
+    // write out new root config so service is bootstrapped (respect white space)
+    fs.writeFile(feathersConfigPath, JSON.stringify(newFeathersConfig, null, 2), function (err) {
+      if (err) {
+        debug(err.stack);
+        return done(err);
+      }
+      debug(`Successfully mounted "${options.name}" service to feathers at ${feathersConfigPath}`);
+      debug(`Service config can be found at ${relativeServiceConfigPath}`);
+      done();
+    });
+  };
+}
+
+export function hooks (options) {
+  return function mount (files, metalsmith, done) {
+    const metadata = metalsmith.metadata();
+
+    let serviceConfigPath = path.resolve(options.mount);
+    let serviceConfigDirname = path.dirname(serviceConfigPath);
+
+    let relativeServiceConfigPath = path.relative(serviceConfigDirname, serviceConfigPath);
+    let existingServiceConfig = require(serviceConfigPath);
+    let serviceConfigChanges = {};
+
+    debug(`Attempting to mount ${options.name} hook to service at ${serviceConfigPath}`);
+
+    metadata.answers.binding.map((b) => {
+      debug(`Compiling changes for ${b} bindings`);
+
+      if (typeof serviceConfigChanges[b] === 'undefined') {
+        serviceConfigChanges[b] = {};
+      }
+
+      metadata.answers.method.map((m) => {
+        debug(`Compiling changes for ${b} bindings and ${m} method`);
+
+        if (typeof serviceConfigChanges[b][m] === 'undefined') {
+          serviceConfigChanges[b][m] = [];
+        }
+
+        let hook = {
+          require: './hooks/' + options.name + '.js',
+          options: []
+        };
+
+        serviceConfigChanges[b][m].push(hook);
+      });
+    });
+
+    debug('Proposed service config changes', serviceConfigChanges);
+    let newServiceConfig = merge(existingServiceConfig, serviceConfigChanges);
+    debug('Final service config to be written', newServiceConfig);
+
+    // write out new root config so service is bootstrapped (respect white space)
+    fs.writeFile(serviceConfigPath, JSON.stringify(newServiceConfig, null, 2), function (err) {
+      if (err) {
+        debug(err.stack);
+        return done(err);
+      }
+      debug(`Successfully mounted "${options.name}" hook to service at ${serviceConfigPath}`);
+      debug(`Service config can be found at ${relativeServiceConfigPath}`);
+      done();
+    });
+
+    done();
+  };
```

Comment:
Can probably refactor a bunch of this to be more streamlined, but don't want to do too much of that until I've laid out the happy path for all of the different generations.